src/components/chat/chat_messages.jsx

```diff
@@ -27,21 +25,23 @@ const ChatMessages = React.createClass({
     }
   },
 
-  componentWillReceiveProps() {
+  componentWillReceiveProps(p) {
     let previousChannel = this.state.channel;
+
     this.setState({
       channel: this.getParams().roomSlug
     }, () => {
-      if (previousChannel !== this.state.channel) {
-        ChatActions.destroy();
-        ChatActions.init(
-          `${AppStore.getUrl()}/rooms/${this.state.channel}/messages`,
-          CurrentUserStore.getToken()
-        );
-      }
+      this.updateMessages();
+      this.handleChannelChange()
     });
   },
 
+  handleChannelChange(){
```

Comment:
I think we'd subscribe them to the room first, and then put them in it -- just automatically.